### PR TITLE
Move ICore/ControlSettings to TerminalControl project

### LINKED ISSUE #7140: Move `TerminalSettings` interfaces
# Description of the new feature/enhancement

- Move `TerminalSettings::ICoreSettings` to TerminalControl project
- Move `TerminalSettings::IControlSettings` to TerminalControl project

`TerminalSettings` will continue to be an `ICoreSettings` and `IControlSettings` and will be used normally.

See #885 and #6904 for more details.

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request
Move `ICoreSettings` and `IControlSettings` from the TerminalSettings project to the TerminalCore and TerminalControl projects respectively. Also entirely removes the TerminalSettings project.

The purpose of these interfaces is unchanged. `ICoreSettings` is used to instantiate a terminal. `IControlSettings` (which requires an `ICoreSettings`) is used to instantiate a UWP terminal control.

## References
Closes #7140 
Related Epic: #885 
Related Spec: #6904 

## PR Checklist
* [X] Closes #7140 
* [X] CLA signed
* [X] Tests ~added~/passed (no additional tests necessary)
* [X] ~Documentation updated~
* [X] ~Schema updated~

## Detailed Description of the Pull Request / Additional comments
A lot of the work here was having to deal with winmd files across all of these projects. The TerminalCore project now outputs a Microsoft.Terminal.TerminalControl.winmd. Some magic happens in TerminalControl.vcxproj to get this to work properly.

## Validation Steps Performed
Deployed Windows Terminal and opened a few new tabs.

### CODE REVIEW

**Comment by carlos-zamora** [comment]
## This is a DRAFT because of a few issues
- [ ] Windows Terminal successfully builds and deploys on a second attempt, not first
- [ ] Linker issues with TerminalApp local tests
- [ ] ICoreSettings/IControlSettings are a part of the `Microsoft::Terminal::TerminalControl` namespace. I think this feels a little awkward. Should we introduce a separate namespace within the TerminalControl project? But that also feels awkward.

File: src/cascadia/TerminalControl/TSFInputControl.cpp [hunk 1]
```diff
@@ -11,9 +11,13 @@ using namespace winrt::Windows::Foundation;
 using namespace winrt::Windows::Graphics::Display;
 using namespace winrt::Windows::UI::Core;
 using namespace winrt::Windows::UI::Text;
-using namespace winrt::Windows::UI::Text::Core;
 using namespace winrt::Windows::UI::Xaml;
 
+namespace winrt
```
[the remark this answers is not in the document]
Comment (carlos-zamora) [reply src/cascadia/TerminalControl/TSFInputControl.cpp:16]: Ah. In a previous iteration, I was exposing `winrt::Microsoft::Terminal::Core`. So every time I used `Core::CoreTextServicesManager` it thought I meant `Microsoft::Terminal::Core` not `Windows::UI::Text::Core`

File: src/cascadia/TerminalApp/CommandPalette.cpp [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #include "CommandPalette.h"
 
 #include "CommandPalette.g.cpp"
-#include <winrt/Microsoft.Terminal.Settings.h>
+#include <winrt/Microsoft.Terminal.TerminalControl.h>
```
[the remark this answers is not in the document]
Comment (DHowett) [reply src/cascadia/TerminalApp/CommandPalette.cpp:8]: Agreed. If we're gonna depend hard on it just throw it in pch.

File: src/cascadia/TerminalControl/TSFInputControl.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (DHowett) [reply src/cascadia/TerminalControl/TSFInputControl.cpp:16]: Ah so this can be backed out?

File: src/cascadia/TerminalCore/Terminal.cpp [hunk 3]
```diff
@@ -344,7 +344,7 @@ void Terminal::UpdateSettings(winrt::Microsoft::Terminal::Settings::ICoreSetting
 
     // If the old scrolloffset was 0, then we weren't scrolled back at all
     // before, and shouldn't be now either.
-    _scrollOffset = originalOffsetWasZero ? 0 : ::base::ClampSub(_mutableViewport.Top(), newVisibleTop);
+    _scrollOffset = originalOffsetWasZero ? 0 : static_cast<int>(::base::ClampSub(_mutableViewport.Top(), newVisibleTop));
```
Comment (DHowett) [inline src/cascadia/TerminalCore/Terminal.cpp:347]: this one seems weird. should we just be usinjg `ClampSub<int>` or something?

File: src/cascadia/TerminalCore/lib/terminalcore-lib.vcxproj [hunk 4]
```diff
@@ -40,20 +47,24 @@
...
       -->
-      <AdditionalIncludeDirectories>$(WinRT_IncludePath)\..\cppwinrt\winrt;"$(OpenConsoleDir)src\cascadia\TerminalSettings\Generated Files";%(AdditionalIncludeDirectories);</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(WinRT_IncludePath)\..\cppwinrt\winrt;"$(OpenConsoleDir)src\cascadia\TerminalCore\Generated Files";%(AdditionalIncludeDirectories);</AdditionalIncludeDirectories>
     </ClCompile>
   </ItemDefinitionGroup>
 
-    <!-- Careful reordering these. Some default props (contained in these files) are order sensitive. -->
+  <!-- Careful reordering these. Some default props (contained in these files) are order sensitive. -->
+  <Import Project="$(OpenConsoleDir)src\cppwinrt.build.post.props" />
   <Import Project="$(SolutionDir)src\common.build.post.props" />
```
Comment (DHowett) [inline src/cascadia/TerminalCore/lib/terminalcore-lib.vcxproj:69]: remove this. cppwinrt includes common already.

File: src/cascadia/TerminalCore/lib/terminalcore-lib.vcxproj [hunk 5]
```diff
@@ -7,16 +7,23 @@
...
 
   <!-- Imported WinRT generated files must go up here to get excluded from Audit correctly. -->
   <PropertyGroup Condition="'$(Configuration)'=='AuditMode'">
-    <CAExcludePath>"$(SolutionDir)\src\cascadia\TerminalSettings\Generated Files\winrt";$(SolutionDir)src\cascadia\TerminalSettings;$(CAExcludePath)</CAExcludePath>
+    <CAExcludePath>"$(SolutionDir)\src\cascadia\TerminalCore\Generated Files\winrt";$(SolutionDir)src\cascadia\TerminalCore;$(CAExcludePath)</CAExcludePath>
   </PropertyGroup>
-    
+
   <Import Project="..\..\..\..\common.openconsole.props" Condition="'$(OpenConsoleDir)'==''" />
   <Import Project="$(SolutionDir)src\common.build.pre.props" />
```
Comment (DHowett) [inline src/cascadia/TerminalCore/lib/terminalcore-lib.vcxproj:25]: remove this. cppwinrt includes common already.

File: src/inc/test/CommonState.hpp [hunk 6]
```diff
@@ -303,7 +303,7 @@ class CommonState
         const CONSOLE_INFORMATION& gci = Microsoft::Console::Interactivity::ServiceLocator::LocateGlobals().getConsoleInformation();
         // length 80 string of text with bisecting characters at the beginning and end.
         // positions of き(\x304d) are at 0, 27-28, 39-40, 67-68, 79
-        PWCHAR pwszText =
+        auto pwszText =
```
Comment (DHowett) [inline src/inc/test/CommonState.hpp:306]: interesting. not necessary for this change, i presume, but I'll allow it? 😁

File: src/cascadia/TerminalCore/Terminal.cpp (hunk 3 above)
Comment (carlos-zamora) [reply src/cascadia/TerminalCore/Terminal.cpp:347]: For whatever reason, this came up:
```
1>D:\Terminal\src\cascadia\TerminalCore\Terminal.cpp(347,104): error C2445: result type of conditional expression is ambiguous: types 'int' and 'base::internal::ClampedNumeric<short>' can be converted to multiple common types
```
It's also having trouble converting from `ClampedNumeric` to the non-clamped numeric type :/

File: src/inc/test/CommonState.hpp (hunk 6 above)
Comment (carlos-zamora) [reply src/inc/test/CommonState.hpp:306]: Similar to the ClampSub issue above, idk why this came up:
```
6>D:\Terminal\src\inc\test\CommonState.hpp(306,25): error C2440: 'initializing': cannot convert from 'const wchar_t [81]' to 'PWCHAR'
6>D:\Terminal\src\inc\test\CommonState.hpp(306,25): message : Conversion from string literal loses const qualifier (see /Zc:strictStrings)
```
🤷‍♂️

File: src/cascadia/TerminalApp/CommandPalette.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlos-zamora) [reply src/cascadia/TerminalApp/CommandPalette.cpp]: Removed most of them. Now we just use it in `KeyChordSerialization` and `TerminalSettingsSerializationHelpers`. Should I still move it to the pch even though it's just 2 files now?

File: src/cascadia/TerminalControl/TSFInputControl.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (carlos-zamora) [reply src/cascadia/TerminalControl/TSFInputControl.cpp]: I _could_ back out the changes to this file, but I left 3 little ones that wouldn't hurt. Can we just ship it as is?

[the remark this answers is not in the document]
Comment (DHowett) [reply src/cascadia/TerminalControl/TSFInputControl.cpp]: i don't mind.

File: src/cascadia/TerminalCore/Terminal.cpp (hunk 3 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalCore/Terminal.cpp:347]: This is bizarre that this is needed now

File: src/cascadia/TerminalApp/CommandPalette.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/CommandPalette.cpp]: Nah this is good.

**Pull request merged**